### PR TITLE
Non-root user for most of the CI images

### PR DESCRIPTION
**Pull request by rcny**
Relates to paritytech/ci_cd#92.

### CODE REVIEW

File: dockerfiles/base-ci-linux/Dockerfile [hunk 1]
```diff
@@ -59,7 +59,10 @@ RUN set -eux; \
 # apt clean up
 	apt-get autoremove -y; \
 	apt-get clean; \
-	rm -rf /var/lib/apt/lists/*
+  rm -rf /var/lib/apt/lists/*; \
+# add non-root user
+  groupadd -g 1000 nonroot; \
+  useradd -u 1000 -g 1000 -s /bin/bash -m nonroot
```
Comment (TriplEight) [inline dockerfiles/base-ci-linux/Dockerfile:65]: both `base-ci-linux` and `base-ci` are designed to be used as the base layers for the other images, other than that they are not used.
So AFAIU you do the preparation part here (for DRY sakes) and actual user's called from this boilerplate in the end images, am I right?

File: dockerfiles/base-ci/Dockerfile [hunk 2]
```diff
@@ -62,7 +62,10 @@ RUN set -eux; \
 # apt clean up
 	apt-get autoremove -y && \
 	apt-get clean && \
-	rm -rf /var/lib/apt/lists/*
+  rm -rf /var/lib/apt/lists/* && \
+# add non-root user
+  groupadd -g 1000 nonroot && \
+  useradd -u 1000 -g 1000 -s /bin/bash -m nonroot
```
Comment (TriplEight) [inline dockerfiles/base-ci/Dockerfile:68]: I've read somewhere that it's generally safer to use ID > 10k to not to accidentally collapse with something, does it worth it?

Another source of "best practices" was even claiming that IDs are obsolete, are you playing safe here or is it better this way?

File: dockerfiles/base-ci-linux/Dockerfile (hunk 1 above)
Comment (rcny) [reply dockerfiles/base-ci-linux/Dockerfile:65]: Yes, that's right.

File: dockerfiles/base-ci/Dockerfile (hunk 2 above)
Comment (TriplEight) [reply dockerfiles/base-ci/Dockerfile:68]: This is valid because we output caches and artifacts that are reused by the other containers and CI processes.

**Review by TriplEight — APPROVED** [review]

**Comment by TriplEight** [comment]
I think this deserves a little doc.

**Pull request merged**